src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AReactor.java

```diff
@@ -299,10 +299,11 @@ public double generateEnergy(final Location l, SlimefunItem sf, Config data) {
 					extraTick(l);
 					int timeleft = progress.get(l);
 					if (timeleft > 0) {
-						if (ChargableBlock.getMaxCharge(l) - ChargableBlock.getCharge(l) >= getEnergyProduction()) {
+						boolean should_charge = ChargableBlock.getMaxCharge(l) - ChargableBlock.getCharge(l) >= getEnergyProduction();
+						if (should_charge) {
 							ChargableBlock.addCharge(l, getEnergyProduction());
 						}
-						if (ChargableBlock.getMaxCharge(l) - ChargableBlock.getCharge(l) >= getEnergyProduction() || !BlockStorage.getBlockInfo(l, "reactor-mode").equals("generator")) {
+						if (should_charge || !BlockStorage.getBlockInfo(l, "reactor-mode").equals("generator")) {
```

Comment:
Apparently this caused some problems.
This shouldn't have changed. Because if should_charge is true, then charge is added, making the charge maybe go completely full capacity.

And this statement should check for that, but since it uses the same boolean as before, this is not taken into account and Reactors wrap around to start at 0 when they are full or burn items that are not needed.

Just letting you know that.

Reply:
This change was to fix Reactors not burning fuel when they should have - if the first if statement was executed then it meant that one unit of energy had been created. By changing the second if to be based on the same condition it makes sure that if there is a charge then the fuel is taken away. For example a reactor that has a storage capacity of 1000J, and a charge rate of 100J/step the following could happen if they are not the same condition:

Step 1
- charge is at 1000J
- nothing happens
Step 2
- Something in the energy network uses 50J of power
- The remaining energy is 950J -> neither if statement executes.
Step 3
- Something in the energy network uses 50J of power
- The remaining energy is 900J -> 
- The first if statement executes, the remainig energy is now 950J.
- The second if does not execute so no time is taken from the fuel
Step 4
- Something in the energy network uses 50J of power
- The remaining energy is 900J -> 
- The first if statement executes, the remainig energy is now 950J.
- The second if does not execute so no time is taken from the fuel

etc., where step 4 is the same as step 3 which means that it can loop infinitely using an infinite amount of energy without using any time remaining on a fuel. You can see this in that in an energy system that is full with a reactor where it will keep the fuel time remaining at the highest value possible.

Reply:
I see, you are right.

Still, how is #666 caused then?
Also ignore my rushed approach then.

Reply:
I think this is more complicated, relating to the algorithm used for energy generation. I'll post in issue #666, can you re-open it?

Reply:
Did so.